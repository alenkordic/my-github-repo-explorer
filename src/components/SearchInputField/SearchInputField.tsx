import React, { useState, useEffect } from "react";
import { TextField } from "@mui/material";

interface SearchInputFieldProps {
  value: string;
  onChange: (s: string) => any;
}

const SearchInputField = ({ onChange, value }: SearchInputFieldProps) => {
  const [searchString, setSearchString] = useState<string>("");

  useEffect(() => {
    const handler = setTimeout(() => {
      if (searchString !== value) {
        onChange(searchString);
      }
    }, 1000);

    return () => {
      clearTimeout(handler);
    };
    // eslint-disable-next-line
  }, [searchString]);

  const onChangeHandler = (e: React.ChangeEvent<HTMLInputElement>): void => {
    setSearchString(e.target.value);
  };

  return (
    <TextField
      fullWidth
      label="Search"
      placeholder="Github repository name"
      variant="outlined"
      value={searchString}
      onChange={onChangeHandler}
    />
  );
};

export default SearchInputField;
