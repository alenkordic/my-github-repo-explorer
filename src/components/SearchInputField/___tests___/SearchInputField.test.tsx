import React from "react";
import SearchInputField from "../SearchInputField";
import { render, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";


it("SearchInput renders correctly", ()=>{
    const { getByPlaceholderText} = render(<SearchInputField onChange={jest.fn()} value=""/>)
    expect(getByPlaceholderText("Github repository name")).toBeTruthy()
})

describe("Input value", ()=> {
    it("updates on change", () =>{
        const {getByPlaceholderText} = render(<SearchInputField onChange={jest.fn()} value=""/>)
        const searchInput = (getByPlaceholderText("Github repository name")as HTMLInputElement)

        fireEvent.change(searchInput, { target: {value: "test"}})
        expect(searchInput.value).toBe("test")
    })
})

//Snapshots
it("Logout btn render correctly", () => {
    const tree = renderer.create(<SearchInputField onChange={jest.fn()} value=""/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  