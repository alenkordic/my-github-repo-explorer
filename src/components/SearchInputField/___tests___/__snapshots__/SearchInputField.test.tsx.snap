// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout btn render correctly 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
>
  <label
    className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
    data-shrink={false}
  >
    Search
  </label>
  <div
    className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Github repository name"
      required={false}
      type="text"
      value=""
    />
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-173wfxe"
      >
        <span>
          Search
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
